packages/wallets/src/features/cashier/flows/WalletDeposit/WalletDeposit.tsx

```diff
@@ -1,10 +1,51 @@
-import React from 'react';
-import { useActiveWalletAccount } from '@deriv/api-v2';
+import React, { useEffect } from 'react';
+import { useActiveWalletAccount, useAuthorize, useCashierFiatAddress, useDepositCryptoAddress } from '@deriv/api-v2';
+import { Loader } from '../../../../components';
+import { isServerError } from '../../../../utils/utils';
 import { DepositCryptoModule, DepositFiatModule } from '../../modules';
+import { DepositErrorScreen } from '../../screens';
 
 const WalletDeposit = () => {
     const { data } = useActiveWalletAccount();
+    const { isSuccess: isAuthorizeSuccess } = useAuthorize();
+    const {
+        error: depositFiatError,
+        isLoading: isFiatAddressLoading,
+        mutate: mutateDepositFiat,
+    } = useCashierFiatAddress();
+    const {
+        error: depositCryptoError,
+        isLoading: isCryptoAddressLoading,
+        mutate: mutateDepositCrypto,
+    } = useDepositCryptoAddress();
+
+    const isLoading = isFiatAddressLoading || isCryptoAddressLoading;
     const isCrypto = data?.currency_config?.is_crypto;
+    const depositError = isCrypto ? depositCryptoError?.error : depositFiatError?.error;
+
+    const getErrorTitle = (errorCode: string) => {
+        if (errorCode === 'CryptoSuspendedCurrency') return 'Error';
+        //TODO: add check for crypto node down
+        return undefined;
+    };
+
+    const errorTitle = depositError ? getErrorTitle(depositError.code) : undefined;
+
+    useEffect(() => {
+        if (isAuthorizeSuccess) {
+            if (isCrypto) {
+                mutateDepositCrypto();
```

Comment:
is there any reason why we use mutate for fetching the data?

Reply:
When loading this component we'd have to fetch the fiat/crypto address to see whether the deposit content should be rendered or if error is returned and we should render the error screen instead

Reply:
ah, I see, the hooks were constructed for this model to begin with ;-/ quite weird for fetching data, but fair enough,